### PR TITLE
Added docstring to the SchedulerType class

### PR DESCRIPTION
**Pull request by Arunprakash-A**
# What does this PR do?
It adds the docstring for the class `SchedulerType` . Currently, the AutoDoc only displays the `__call__ ` signature of `enum.EnumType` base class which is not helpful.  One needs to look for the `names` and the respective mapping in the source code of `trainer.py`. Therefore, added the docstring that lists out all the available names and the scheduler types they are mapped to. 

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case)

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Documentation: @stevhliu

### CODE REVIEW

File: src/transformers/trainer_utils.py [hunk 1]
```diff
@@ -411,6 +411,21 @@ def speed_metrics(split, start_time, num_samples=None, num_steps=None, num_token
...
+    "linear" = get_linear_schedule_with_warmup
+    "cosine" = get_cosine_schedule_with_warmup
+    "cosine_with_restarts" = get_cosine_with_hard_restarts_schedule_with_warmup
+    "polynomial" = get_polynomial_decay_schedule_with_warmup
+    "constant" =  get_constant_schedule
+    "constant_with_warmup" = get_constant_schedule_with_warmup
+    "inverse_sqrt" = get_inverse_sqrt_schedule
+    "reduce_lr_on_plateau" = get_reduce_on_plateau_schedule
+    "cosine_with_min_lr" = get_cosine_with_min_lr_schedule_with_warmup
+    "warmup_stable_decay" = get_wsd_schedule
```
Comment (stevhliu) [inline src/transformers/trainer_utils.py:427]:
```suggestion
    Scheduler names for the parameter `lr_scheduler_type` in [`TrainingArguments`].
    By default, it uses "linear". Internally, this retrieves `get_linear_schedule_with_warmup` scheduler from [`Trainer`].
    
    Scheduler types:
       - "linear" = get_linear_schedule_with_warmup
       - "cosine" = get_cosine_schedule_with_warmup
       - "cosine_with_restarts" = get_cosine_with_hard_restarts_schedule_with_warmup
       - "polynomial" = get_polynomial_decay_schedule_with_warmup
       - "constant" =  get_constant_schedule
       - "constant_with_warmup" = get_constant_schedule_with_warmup
       - "inverse_sqrt" = get_inverse_sqrt_schedule
       - "reduce_lr_on_plateau" = get_reduce_on_plateau_schedule
       - "cosine_with_min_lr" = get_cosine_with_min_lr_schedule_with_warmup
       - "warmup_stable_decay" = get_wsd_schedule
```

**Review by stevhliu — COMMENTED** [review]
Thanks for expanding this! Just a few formatting issues and should be good to merge 🤗

**Comment by stevhliu** [comment]
Thanks! Can you implement the fix suggested from the CI test [here](https://app.circleci.com/pipelines/github/huggingface/transformers/101290/workflows/6c09ee5a-751f-46ca-b50f-4451a46219c9/jobs/1349645) and then run `make style` after?

**Comment by Arunprakash-A** [comment]
> Thanks! Can you implement the fix suggested from the CI test [here](https://app.circleci.com/pipelines/github/huggingface/transformers/101290/workflows/6c09ee5a-751f-46ca-b50f-4451a46219c9/jobs/1349645) and then run `make style` after?

I removed the blank line at 417. However,  I have no clue about running `make style` ?  Could you help me with that?

**Comment by stevhliu** [comment]
Yeah, you can run `make style` from the command line

**Comment by Arunprakash-A** [comment]
> Yeah, you can run `make style` from the command line

It passed all checks.

``` 
make style
ruff check examples tests src utils setup.py conftest.py --fix --exclude ""
All checks passed!
ruff format examples tests src utils setup.py conftest.py --exclude ""
3054 files left unchanged
make autogenerate_code
make[1]: Entering directory '/home/arun/transformers'
running deps_table_update
updating src/transformers/dependency_versions_table.py
make[1]: Leaving directory '/home/arun/transformers'
make extra_style_checks
make[1]: Entering directory '/home/arun/transformers'
python utils/custom_init_isort.py
python utils/sort_auto_mappings.py
python utils/check_doc_toc.py --fix_and_overwrite
make[1]: Leaving directory '/home/arun/transformers'
```
